scaleway/resource_instance_security_group.go

```diff
@@ -205,18 +234,28 @@ func resourceScalewayInstanceSecurityGroupUpdate(d *schema.ResourceData, m inter
 		return err
 	}
 
-	// *******
-	// Handle SecurityGroupRules
-	//
-	// It works as followed:
-	//   1) Creates 2 map[direction][]rule: one for rules in state and one for rules in API
-	//   2) For each direction we:
-	//     A) Loop for each rule in state for this direction
-	//       a) Compare with api rule in this direction at the same index
-	//          if different update / if equals do nothing / if no more api rules to compare create new api rule
-	//     B) If there is more rule in the API we remove them
-	// *******
+	if d.Get("external_rules") != nil && d.Get("external_rules").(bool) {
```

Comment:
why do you check nil?

Reply:
tests were failing: `external_rules` is optionnal